spec/amp-cors-requests.md

```diff
@@ -37,7 +37,9 @@ by default.
 
 ## CORS Security in AMP
 
-This security protocol consists of two components: CORS origin and source origin restrictions.
+This security protocol consists of three components: `AMP-Same-Origin`, CORS origin and source origin restrictions.
+
+On same origin requests, AMP will set `AMP-Same-Origin: true` custom header. If this header is set it indicates the request is coming from same origin.
```

Comment:
@molnarg @mkhatib Here's why I believe we need to check `Origin` first: this is all due to preflight. Some, possibly sloppy or possibly valid reason, implementation of preflight could allow a wide range of CORS requests. Then, a CORS request could set `AMP-Same-Origin: true` header via `Access-Control-Request-Headers`. I realize this is pretty far-fetched and doesn't sound like it'd be supported by any valid use case. So, if you don't think this is a valid reason, let's check `AMP-Same-Origin` first - it'd certainly be simpler.